Src/LiquidProjections.Testing/EnumerableExtensions.cs

```diff
@@ -19,10 +19,7 @@ public static IEnumerable<IList<T>> InBatchesOf<T>(this IEnumerable<T> items, in
                 }
             }
 
-            if (batch.Count != 0)
-            {
-                yield return batch;
-            }
+            yield return batch;

```

Comment:
I don't see why you want to make that a responsibility of this method. Its name tells us that it should just split the sequence into batches of the given size.

Also the current implementation could lead to 3 different scenarios:

1. It can return 1 empty batch
2. It can return 1 or more non-empty batches
3. It can return 1 or more non-empty batches followed by an empty batch